docs/addons/external-secrets.md

```diff
@@ -0,0 +1,98 @@
+# External Secrets Add-On
+
+External Secrets add-on is based
+on [External Secrets Operator (ESO)](https://github.com/external-secrets/external-secrets) and allows integration with
+third-party secret stores like AWS Secrets Manager and inject the values into the EKS cluster as Kubernetes Secrets
+
+## Usage
+
+```typescript
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.ExternalsSecretsAddOn({});
+
+const blueprint = blueprints.EksBlueprint.builder()
+    .addOns(addOn)
+    .build(app, 'my-stack-name');
+```
+
+## Cluster Secret Store
+
+Create a [ClusterSecretStore](https://external-secrets.io/v0.5.9/api-clustersecretstore/) which can be used by all
+ExternalSecrets from all namespaces
+
+```typescript
+import * as eks from 'aws-cdk-lib/aws-eks';
+
+const cluster = blueprint.getClusterInfo().cluster;
+
+const clusterSecretStore = new eks.KubernetesManifest(scope, "ClusterSecretStore", {
```

Comment:
Does it make sense to create this resource as part of the add-on deployment? It sounds like it may be useless without it and developers won't know which sa to use. Another option to is create it as part of the team structure. That will work if you envision more than a single ClusterSecretStore in a cluster as a common use case.

Reply:
I had similiar thoughts. But without wildcard permissions.

For the Teams Feature the namespaced SecretStore could be a better fit.